packages/shared/lib/integrations/scripts/webhook/linear-webhook-routing.ts

```diff
@@ -0,0 +1,40 @@
+import crypto from 'node:crypto';
+
+import type { Config as ProviderConfig } from '../../../models/Provider.js';
+import type { WebhookHandler } from './types.js';
+
+interface LinearBody {
+    action: string;
+    data: Record<string, unknown>;
+    type: string;
+    createdAt: string;
+}
+
+function validate(integration: ProviderConfig, headerSignature: string, rawBody: string): boolean {
+    if (!integration.custom?.['webhookSecret']) {
+        return false;
+    }
+
+    const signature = crypto.createHmac('sha256', integration.custom['webhookSecret']).update(rawBody).digest('hex');
+    return signature === headerSignature;
```

Comment:
Should use `timingSafeEqual` instead a la: https://github.com/NangoHQ/nango/blob/master/packages/shared/lib/integrations/scripts/webhook/hubspot-webhook-routing.ts#L11-L15

Reply:
good call